src/app/productConfig/productConfigForm/productConfigForm.component.js

```diff
@@ -388,6 +389,11 @@ class ProductConfigFormController {
       this.$window.location = url
     }
   }
+
+  shouldShowForcedUserToLogoutError() {
```

Comment:
Tell me more about this new function

Reply:
On adding an item to the cart, we check to see if the user has a `public` role, is logged into Okta and has no items in their account. If so, we automatically force the user to logout.

This was happening before, but when it was logging out, it didn't require a refresh. Okta requires a refresh since we send the user to the Okta site to clear Okta's session data and then return to the Give site.

We handle this here `src/common/services/api/cart.service.js` - Lines 164 - 181

On the `sessionService.oktaSignOut` we pass a prop which tells session data to store a prop called `forcedUserToLogout`. We also ensure Okta returns the user to the same page.
`src/common/services/session/session.service.js` Lines 317 - 322

On`src/app/productConfig/productConfigForm/productConfigForm.component.js`, we watch for the `forcedUserToLogout` session data and show the warning to the user if present.

Reply:
Thanks!